src/core/lib/security/transport/security_handshaker.c

```diff
@@ -193,8 +248,12 @@ static grpc_error *check_peer_locked(grpc_exec_ctx *exec_ctx,
     return grpc_set_tsi_error_result(
         GRPC_ERROR_CREATE("Peer extraction failed"), result);
   }
-  grpc_security_connector_check_peer(exec_ctx, h->connector, peer,
-                                     &h->auth_context, &h->on_peer_checked);
+  grpc_security_connector *lb_channel_connector =
+      find_lb_connector(exec_ctx, h->args->args);
```

Comment:
If we do wind up creating a new security connector here, don't we need to unref it before we return from here?

(This comment is irrelevant if you take my suggestion above about not creating a new security connector to begin with.)